### ISSUE TITLE
ListView dynamic itemCount shows error Rendering

### ISSUE THREAD

**Issue by Dakhtara** [flutter/flutter#20256]
## Steps to Reproduce

It's a small application that has a listView.builder with a TextField.
It lists all Artists available from an API and I can query to find Artist by name.

The pagination works well, when I write in the TextField, it reloads well the list. But when it comes back to the whole list (just delete text from TextField) I have this logs that crash the app.

UPDATE: it only crash when the Textfield has searched for items, then I empty the Textfield so it reloads the ListView with 20 items and it crash when I scroll to load the 2nd page of items

```
@override
  Widget build(BuildContext context) {
    // TODO: implement build
    return new Container(
      child: new ListView.builder(
        itemCount: artists.length + 1,
        controller: _onScrollBottom,
        itemBuilder: (BuildContext context, int index) {
          if (index == 0) {
            return new Container(
                child: new Column(
              children: <Widget>[
                new Text("Liste des artistes  $term - ${artists.length}"),
                new Padding(
                  padding: EdgeInsets.all(16.0),
                  child: new TextField(
                    controller: _searchQuery,
                  ),
                )
              ],
            ));
          }
          index -= 1;

          return this._artistTile(artists[index]);
        },
      ),
    );
  }
 //Lock la requete pour éviter des requetes multiples
  // Voir https://github.com/fablue/building-a-social-network-with-flutter#lockedloadnext
  void lockedLoadArtist({String q: '', newPage: false}) {
    if (this.request == null && (q == '' || q != term)) {
      if (q != term && newPage == false) {
        _page = 1;
        hasMorePage = true;
      } else if (hasMorePage == true && q == term) {
        _page++;
      } else {
        return;
      }

      print('term: $q, page: $_page');
      this.request =
          ArtistRepository().fetchArtists(page: _page, term: q).then((x) {
        this.request = null;
        loadMoreStatus = LOAD_MORE_STATUS.STABLE;
        if (q != term) {
          artists.clear();
        }
        setState(() {
          term = q;
          artists.addAll(x);
          if (artists.length % 20 != 0) {
            hasMorePage = false;
          }
        });
      });
    }
  }
```
Sorry for horrible code, I try alot of solution before posting. But I'm really stuck for the moment.
## Logs
I understand the output logs, that it try to rerender but i have no idea of what to do.
```
I/flutter (29875): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (29875): The following assertion was thrown building NotificationListener<KeepAliveNotification>:
I/flutter (29875): 'package:flutter/src/rendering/object.dart': Failed assertion: line 2775 pos 14:
I/flutter (29875): '_debugUltimatePreviousSiblingOf(after, equals: _firstChild)': is not true.
I/flutter (29875):
I/flutter (29875): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (29875): more information in this error message to help you determine and fix the underlying cause.
I/flutter (29875): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (29875):   https://github.com/flutter/flutter/issues/new
I/flutter (29875):
I/flutter (29875): When the exception was thrown, this was the stack:
I/flutter (29875): #2      _RenderSliverMultiBoxAdaptor&RenderSliver&ContainerRenderObjectMixin._insertIntoChildList (package:flutter/src/rendering/object.dart)
I/flutter (29875): #3      _RenderSliverMultiBoxAdaptor&RenderSliver&ContainerRenderObjectMixin.insert (package:flutter/src/rendering/object.dart:2809:5)
I/flutter (29875): #4      RenderSliverMultiBoxAdaptor.insert (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:209:11)
I/flutter (29875): #5      SliverMultiBoxAdaptorElement.insertChildRenderObject (package:flutter/src/widgets/sliver.dart:865:18)
I/flutter (29875): #6      RenderObjectElement.attachRenderObject (package:flutter/src/widgets/framework.dart:4513:35)
I/flutter (29875): #7      RenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4279:5)
I/flutter (29875): #8      SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4653:11)
I/flutter (29875): #9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:2907:14)
I/flutter (29875): #9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:2907:14)
I/flutter (29875): #10     Element.updateChild (package:flutter/src/widgets/framework.dart:2710:12)
I/flutter (29875): #11     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #12     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #13     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3622:5)
I/flutter (29875): #14     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3617:5)
I/flutter (29875): #15     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2907:14)
I/flutter (29875): #16     Element.updateChild (package:flutter/src/widgets/framework.dart:2710:12)
I/flutter (29875): #17     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #18     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #19     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3622:5)
I/flutter (29875): #20     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3617:5)
I/flutter (29875): #21     ParentDataElement.mount (package:flutter/src/widgets/framework.dart:3955:11)
I/flutter (29875): #22     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2907:14)
I/flutter (29875): #23     Element.updateChild (package:flutter/src/widgets/framework.dart:2710:12)
I/flutter (29875): #24     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #25     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #26     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3622:5)
I/flutter (29875): #27     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3769:11)
I/flutter (29875): #28     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3617:5)
I/flutter (29875): #29     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2907:14)
I/flutter (29875): #30     Element.updateChild (package:flutter/src/widgets/framework.dart:2710:12)
I/flutter (29875): #31     SliverMultiBoxAdaptorElement.updateChild (package:flutter/src/widgets/sliver.dart:744:36)
I/flutter (29875): #32     SliverMultiBoxAdaptorElement.performRebuild (package:flutter/src/widgets/sliver.dart:702:34)
I/flutter (29875): #33     SliverMultiBoxAdaptorElement.update (package:flutter/src/widgets/sliver.dart:671:7)
I/flutter (29875): #34     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #35     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #36     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #37     ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #38     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #39     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #40     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #41     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:4379:32)
I/flutter (29875): #42     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4769:17)
I/flutter (29875): #43     _ViewportElement.update (package:flutter/src/widgets/viewport.dart:192:11)
I/flutter (29875): #44     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #45     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #46     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #47     ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #48     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #49     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #50     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #51     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #52     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #53     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #54     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #55     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #56     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #57     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #58     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #59     StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #60     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #61     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #62     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #63     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #64     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #65     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #66     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #67     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #68     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #69     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #70     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #71     StatelessElement.update (package:flutter/src/widgets/framework.dart:3702:5)
I/flutter (29875): #72     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #73     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #74     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #75     StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #76     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #77     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #78     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #79     StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #80     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #81     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #82     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #83     StatelessElement.update (package:flutter/src/widgets/framework.dart:3702:5)
I/flutter (29875): #84     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #85     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #86     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #87     StatelessElement.update (package:flutter/src/widgets/framework.dart:3702:5)
I/flutter (29875): #88     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #89     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #90     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #91     StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #92     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #93     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #94     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #95     StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #96     Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #97     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #98     Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #99     ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #100    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #101    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #102    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #103    ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #104    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #105    RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:4379:32)
I/flutter (29875): #106    MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4769:17)
I/flutter (29875): #107    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #108    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #109    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #110    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #111    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #112    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #113    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #114    ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #115    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #116    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #117    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #118    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #119    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #120    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #121    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #122    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #123    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #124    StatelessElement.update (package:flutter/src/widgets/framework.dart:3702:5)
I/flutter (29875): #125    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #126    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #127    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #128    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #129    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #130    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #131    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #132    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #133    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #134    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #135    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #136    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #137    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #138    ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #139    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #140    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #141    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #142    ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #143    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #144    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #145    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #146    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #147    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #148    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #149    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #150    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #151    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #152    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #153    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #154    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #155    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #156    StatelessElement.update (package:flutter/src/widgets/framework.dart:3702:5)
I/flutter (29875): #157    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #158    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #159    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #160    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #161    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #162    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #163    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #164    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #165    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #166    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #167    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #168    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #169    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #170    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #171    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #172    StatelessElement.update (package:flutter/src/widgets/framework.dart:3702:5)
I/flutter (29875): #173    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #174    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #175    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #176    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #177    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #178    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #179    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #180    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #181    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #182    ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #183    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #184    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #185    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #186    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #187    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #188    StatefulElement.update (package:flutter/src/widgets/framework.dart:3799:5)
I/flutter (29875): #189    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #190    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #191    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #192    StatelessElement.update (package:flutter/src/widgets/framework.dart:3702:5)
I/flutter (29875): #193    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #194    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4661:14)
I/flutter (29875): #195    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #196    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #197    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #198    ProxyElement.update (package:flutter/src/widgets/framework.dart:3909:5)
I/flutter (29875): #199    Element.updateChild (package:flutter/src/widgets/framework.dart:2699:15)
I/flutter (29875): #200    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3653:16)
I/flutter (29875): #201    Element.rebuild (package:flutter/src/widgets/framework.dart:3495:5)
I/flutter (29875): #202    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2242:33)
I/flutter (29875): #203    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:626:20)
I/flutter (29875): #204    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:208:5)
I/flutter (29875): #205    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
I/flutter (29875): #206    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
I/flutter (29875): #207    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:842:5)
I/flutter (29875): #208    _invoke (dart:ui/hooks.dart:120:13)
I/flutter (29875): #209    _drawFrame (dart:ui/hooks.dart:109:3)
I/flutter (29875): (elided 2 frames from class _AssertionError)
I/flutter (29875): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter (29875): Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 2775 pos 14: '_debugUltimatePreviousSiblingOf(after, equals: _firstChild)': is not true.
I/chatty  (29875): uid=10478(u0_a478) 1.ui identical 1 line
I/flutter (29875): Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 2775 pos 14: '_debugUltimatePreviousSiblingOf(after, equals: _firstChild)': is not true.
I/flutter (29875): Another exception was thrown: A RenderSliverPadding expected a child of type RenderSliver but received a child of type RenderErrorBox.
I/flutter (29875): Another exception was thrown: 'package:flutter/src/widgets/framework.dart': Failed assertion: line 3497 pos 14: 'owner._debugCurrentBuildTarget == this': is not true.
I/chatty  (29875): uid=10478(u0_a478) 1.ui identical 28 lines
I/flutter (29875): Another exception was thrown: 'package:flutter/src/widgets/framework.dart': Failed assertion: line 3497 pos 14: 'owner._debugCurrentBuildTarget == this': is not true.
```

Flutter doctor
```
[flutter] flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, v0.5.1, on Microsoft Windows [version 10.0.17134.165], locale fr-FR)
[√] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[!] Android Studio (version 2.0)
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Unable to find bundled Java version.
[!] VS Code, 32-bit edition
[!] VS Code, 64-bit edition (version 1.25.1)
[√] Connected devices (1 available)

! Doctor found issues in 3 categories.
```

**Comment by Dakhtara** [comment]
I just add a new message. 

I found what goes wrong. It's when the TextField keep focus and when i scroll to load more items in the ListView. 

So to workaround this, I have added a ScrollController on the ListView and a FocusNode. So when I listen to scroll, I just stop focus on TextField. And it works with this.

But I am still wondering if this error is normal or not. Because if I scroll with the focus on the TextField I always have the error. 

**How to reproduce : **

- Generate a ListView, with a TextField.
- Focus on a TextField
- Scroll down to generate new Items
